### PR TITLE
Upgrade Espresso

### PR DESCRIPTION
**Pull request by seadowg**
This upgrades to the newer version of Espresso which will hopefully make our instrumentation tests more stable. I've given it a few test runs and everything looks green so far.

### CODE REVIEW

File: collect_app/src/androidTest/java/org/odk/collect/android/support/pages/Page.kt [hunk 1]
```diff
@@ -419,7 +419,7 @@ abstract class Page<T : Page<T>> {
                 InstrumentationRegistry.getInstrumentation().runOnMainSync {
                     val activities = ActivityLifecycleMonitorRegistry.getInstance().getActivitiesInStage(Stage.RESUMED)
                     if (!activities.isEmpty()) {
-                        activity[0] = Iterables.getOnlyElement(activities)
+                        activity[0] = Iterables.getOnlyElement(activities) as Activity
```
Comment (seadowg) [inline collect_app/src/androidTest/java/org/odk/collect/android/support/pages/Page.kt:422]: Signature change on `getActivitiesInStage`  requires an explicit cast here.

File: collect_app/src/androidTest/java/org/odk/collect/android/support/rules/CollectTestRule.kt [hunk 2]
```diff
@@ -44,7 +44,7 @@ class CollectTestRule @JvmOverloads constructor(
             .addProject()
 
     fun launchShortcuts(): ShortcutsPage {
-        val scenario = launch(AndroidShortcutsActivity::class.java)
+        val scenario = launchForResult(AndroidShortcutsActivity::class.java)
```
Comment (seadowg) [inline collect_app/src/androidTest/java/org/odk/collect/android/support/rules/CollectTestRule.kt:47]: We need to use `launchFoResult` to use `result` now

File: collect_app/src/androidTest/java/org/odk/collect/android/support/pages/Page.kt [hunk 3]
```diff
@@ -252,7 +252,7 @@ abstract class Page<T : Page<T>> {
     }
 
     fun <D : Page<D>?> rotateToLandscape(destination: D): D {
-        onView(isRoot()).perform(rotateToLandscape())
+        currentActivity!!.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
```
Comment (seadowg) [inline collect_app/src/androidTest/java/org/odk/collect/android/support/pages/Page.kt:255]: This seems to be more reliable than the Espresso action

File: buildSrc/src/main/java/dependencies/Versions.kt [hunk 4]
```diff
@@ -7,7 +7,7 @@ object Versions {
 
     const val androidx_fragment = "1.4.1"
     const val dagger = "2.42"
-    const val espresso = "3.4.0"
+    const val espresso = "3.5.0-beta01"
```
Comment (seadowg) [inline buildSrc/src/main/java/dependencies/Versions.kt:10]: This is still beta, but the releases for Espresso move slowly, and we've been having enough trouble that I think it's worth it.

**Review by grzesiek2010 — APPROVED** [review]

**Pull request merged**